src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -1309,19 +1305,17 @@ private Task<TResult> WriteToOutputAsync<TResult>(ODataPayloadKind payloadKind,
         {
             // Set the content type header here since all headers have to be set before getting the stream
             this.SetOrVerifyHeaders(payloadKind);
-
             // Create the output context
-            return this.message.GetStreamAsync()
-                .FollowOnSuccessWithTask(
-                    streamTask => this.format.CreateOutputContextAsync(
-                        this.GetOrCreateMessageInfo(streamTask.Result, true),
-                        this.settings))
-                .FollowOnSuccessWithTask(
-                    createOutputContextTask =>
-                    {
-                        this.outputContext = createOutputContextTask.Result;
-                        return writeFunc(this.outputContext);
-                    });
+            return WriteToOutputInnerAsync();
+            async Task<TResult> WriteToOutputInnerAsync()

```

Comment:
wh is this method caalled `WriteToOutputInnerAsync` and not `WriteToOutputContextAsync`

Reply:
@ElizabethOkerio The method it belongs to is named `WriteToOutputAsync` so the inner method just adopted that name with `Inner` just before `Async`